src/models/config.py

```diff
@@ -98,6 +113,18 @@ def hashify_all_families(command = {}):
             Config.__copy_values(command, family_hash, family)
         return combined_hash
 
+    # Also generates a similar hash - for node groups
+    #   {
+    #       groupX: **<node>,
+    #       ...
+    #   }
+    def hashify_all_groups(command = {}):
+        combined_hash = {}
+        for group in Config.all_groups():
+            group_hash = combined_hash.setdefault(group, {})
+            Config.__copy_values(command, group_hash, group)
```

Comment:
This `copy` value method is getting used a lot and should be rethought. I think the problem lies with the `hashify*` methods in general. There is probably and underlining structure that can be extracted/ shared.

For the time being, just call the `Copy.__copy_values`  method (even thought it is private); as this shouldn't delay the release. However can you open an issue about it and reference this comment?